src/temperature/crud.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+from datetime import datetime
+
+from httpx import AsyncClient
+from sqlalchemy import select, insert
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import selectinload
+
+from src.city.crud import get_all_cities
+from src.city.models import DBCity
+from src.temperature.models import DBTemperature
+from weather_api import get_api_temperature
+
+
+async def set_cities_temperature(db: AsyncSession):
+    cities = await get_all_cities(db)
+    async with AsyncClient() as client:
+        tasks = [
+            asyncio.create_task(create_or_update_db_temperature(db, city, client))
+            for city in cities
+        ]
+        await asyncio.gather(*tasks)
+
+    await db.commit()
+
+    return await list_cities_temperature(db)
+
+
+async def list_cities_temperature(db: AsyncSession, city_id: int = None):
+    query = select(DBTemperature).options(selectinload(DBTemperature.city))
+    if city_id:
+        query = query.filter(DBTemperature.city_id == city_id)
+
+    result = await db.execute(query)
+
+    return [city[0] for city in result.fetchall()]
+
+
+async def get_db_temperature_by_city_id(db: AsyncSession, city: DBCity):
+    query = select(DBTemperature).where(DBTemperature.city_id == city.id)
+    result = await db.execute(query)
+    temperature = result.fetchall()
+
+    if temperature:
+        return temperature[0][0]
```

Comment:
The return statement `return temperature[0][0]` assumes that `temperature` is a list of tuples with each tuple containing a single element. Ensure this structure is consistent with the query results.